ansible_collections/f5networks/f5_modules/plugins/modules/bigip_policy_rule.py

```diff
@@ -858,21 +870,31 @@ def _handle_http_uri_condition(self, action, item):
             ))
 
     def _handle_tcp_condition(self, action, item):
+        options = ['address_matches_with_any']
         action['type'] = 'tcp'
-        if 'address_matches_with_any' not in item:
+        if all(k not in item for k in options):
             raise F5ModuleError(
-                "A 'address_matches_with_any' must be specified when the 'tcp' type is used."
+                "A 'address_matches_with_any' must be specified "
+                "when the 'tcp' type is used."
             )
-        if isinstance(item['address_matches_with_any'], list):
-            values = item['address_matches_with_any']
-        else:
-            values = [item['address_matches_with_any']]
-        action.update(dict(
-            address=True,
-            matches=True,
-            # clientAccepted=True,
-            values=values
-        ))
+        if 'address_matches_with_any' in item and item['address_matches_with_any'] is not None:
+            if isinstance(item['address_matches_with_any'], list):
+                values = item['address_matches_with_any']
+            else:
+                values = [item['address_matches_with_any']]
+            action.update(dict(
+                address=True,
+                matches=True,
+                values=values
+            ))
+        if 'event' not in item:
```

Comment:
is event needed by the API to create TCP resource if yes then this is fine, just add mention of this into the docs above. If this is not needed then remove this condition not to force users to specify event

Reply:
changed